backend/tests/Notifo.Domain.Tests/Integrations/Discord/DiscordTests.cs

```diff
@@ -0,0 +1,154 @@
+    }
+
+    [Fact]
+    public async Task Send_full_message()
+    {
+        var client = CreateClient();
+
+        var message = new MessagingMessage
+        {
+            Text = "Test message",
+            DetailedBodyText = "Detailed body text",
+        };
+
+        var userId = GetUserId();
+        AddTargets(message, userId);
+        var result = await client.System.SendAsync(client.Context, message, default);
+
+        Assert.Equal(DeliveryResult.Handled, result);
+    }
+
+    [Fact]
+    public async Task Should_failed_on_user_without_app_installed()
+    {
+        var client = CreateClient();
+
+        var message = new MessagingMessage { Text = "Test message" };
+
+        var userId = "261155655006000000"; // Random userId
+        AddTargets(message, userId);
+
+        var result = await client.System.SendAsync(client.Context, message, default);
+
+        Assert.Equal(DeliveryStatus.Failed, result.Status);
+    }
+
+    [Fact]
+    public async Task Should_accept_images()
+    {
+        var client = CreateClient();
+
+        var message = new MessagingMessage
+        {
+            Text = "Test message",
+            ImageSmall = "https://picsum.photos/200/300",
+            ImageLarge = "https://picsum.photos/400/600"
+        };
+
+        var userId = GetUserId();
+        AddTargets(message, userId);
+        var result = await client.System.SendAsync(client.Context, message, default);
+
+        Assert.Equal(DeliveryResult.Handled, result);
+    }
+
+    [Fact]
+    public async Task Should_accept_urls()
+    {
+        var client = CreateClient();
+
+        var message = new MessagingMessage
+        {
+            Text = "Test message",
+            LinkUrl = "https://notifo.io",
+            LinkText = "Notifo"
+        };
+        var userId = GetUserId();
+        AddTargets(message, userId);
+        var result = await client.System.SendAsync(client.Context, message, default);
+
+        Assert.Equal(DeliveryResult.Handled, result);
+    }
+
+    static ResolvedIntegration<DiscordIntegration> CreateClient()
+    {
+        var botToken = TestHelpers.Configuration.GetValue<string>("discord:botToken")!;
+
+        var context = BuildContext(new Dictionary<string, string>
+        {
+            [DiscordIntegration.BotToken.Name] = botToken,
+        });
+
+        var integration =
+            new ServiceCollection()
+                .AddIntegrationDiscord()
+                .AddMemoryCache()
+                .BuildServiceProvider()
+                .GetRequiredService<DiscordIntegration>();
+
+        Assert.NotNull(integration);
+        return new ResolvedIntegration<DiscordIntegration>(Guid.NewGuid().ToString(), context, integration);
+    }
+
+    private string GetUserId()
+    {
+        var id = TestHelpers.Configuration.GetValue<string>("discord:userId");
+
+        Assert.False(string.IsNullOrWhiteSpace(id), "Please set the Discord userId in the testing env");
```

Comment:
I would not have expected that here. Just throw an exception, but do not use asserts for non-assertions.